src/core/ddsi/src/ddsi_ipaddr.c

```diff
@@ -104,7 +104,8 @@ enum ddsi_locator_from_string_result ddsi_ipaddr_from_string (const struct ddsi_
   const size_t len = strlen(str);
   if (len == 0 || len >= sizeof(copy))
     return AFSR_INVALID;
-  memcpy(copy, str, len + 1);
+  memcpy(copy, str, len);
+  copy[len] = '\0';
```

Comment:
I don't really understand why this is needed here (except for keeping the sanitizer quiet), because `strlen` would have a problem when there is no trailing `\0`. Maybe replace these two lines by a `strcpy`?

Reply:
Replaced it by using `ddsrt_strlcpy` instead.